src/compiler/emitter.ts

```diff
@@ -2505,6 +2507,87 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
+                    //                  ^ -> required extra parenthesis controlled by shouldEmitParenthesis
+                    // Otherwise, false
+                    let shouldEmitParenthesis = false;
+
+                    if (isElementAccessExpression(leftHandSideExpression)) {
+                        shouldEmitParenthesis = node.parent.kind === SyntaxKind.VariableDeclaration || node.parent.kind === SyntaxKind.BinaryExpression;
+ 
+                        if (shouldEmitParenthesis) {
+                            write("(");
+                        }
+
+                        synthesizedLHS = <ElementAccessExpression>createSynthesizedNode(SyntaxKind.ElementAccessExpression, /*startsOnNewLine*/ false);
+                        let tempExpression = createAndRecordTempVariable(TempFlags.Auto);
+                        emitAssignment(tempExpression, leftHandSideExpression.expression, /*shouldEmitCommaBeforeAssignment*/ false);
+                        synthesizedLHS.expression = tempExpression
+
+                        if (leftHandSideExpression.argumentExpression.kind !== SyntaxKind.NumericLiteral &&
+                            leftHandSideExpression.argumentExpression.kind !== SyntaxKind.StringLiteral) {
+                            let tempArgumentExpression = createAndRecordTempVariable(TempFlags._i);
+                            (<ElementAccessExpression>synthesizedLHS).argumentExpression = tempArgumentExpression;
+                            emitAssignment(tempArgumentExpression, leftHandSideExpression.argumentExpression, /*shouldEmitCommaBeforeAssignment*/ true);
+                        }
+                        else {
+                            (<ElementAccessExpression>synthesizedLHS).argumentExpression = leftHandSideExpression.argumentExpression;
+                        }
+                        write(", ");
+                    }
+                    else if (isPropertyAccessExpression(leftHandSideExpression)) {
+                        shouldEmitParenthesis = node.parent.kind === SyntaxKind.VariableDeclaration || node.parent.kind === SyntaxKind.BinaryExpression;
+
+                        if (shouldEmitParenthesis) {
+                            write("(");
+                        }
+
+                        synthesizedLHS = <PropertyAccessExpression>createSynthesizedNode(SyntaxKind.PropertyAccessExpression, /*startsOnNewLine*/ false);
+                        let tempExpression = createAndRecordTempVariable(TempFlags.Auto);
+                        synthesizedLHS.expression = tempExpression
+                        emitAssignment(tempExpression, leftHandSideExpression.expression,  /*shouldEmitCommaBeforeAssignment*/ false);
+                        (<PropertyAccessExpression>synthesizedLHS).dotToken = leftHandSideExpression.dotToken;
+                        (<PropertyAccessExpression>synthesizedLHS).name = leftHandSideExpression.name;

```

Comment:
They are two lines that are same. though I cant refactor that out unless I have `if` clause to cover everything